src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2429,209 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
+        let file_span = (|| {
+            let source_map = tcx.sess.source_map();
+            let crate_src = tcx.sess.local_crate_source_file.as_ref()?;
+            let abs_crate_src = crate_src.canonicalize().ok()?;
+            let crate_root = abs_crate_src.parent()?.parent()?;
+            let rel_path = path.strip_prefix(crate_root).ok()?;
+            let files = source_map.files();
+            let file = files.iter().find(|file| match &file.name {
+                FileName::Real(RealFileName::LocalPath(other_path)) => rel_path == other_path,
+                _ => false,
+            })?;
+            Some(rustc_span::Span::with_root_ctxt(
+                file.start_pos + BytePos(byte_offset),
+                file.start_pos + BytePos(byte_ceiling),
+            ))
+        })()
+        .unwrap_or(rustc_span::DUMMY_SP);
+
+        // The root path is the inverse of Context::current
+        let root_path = vec!["../"; cx.current.len() - 1].join("");
+
+        let mut decoration_info = FxHashMap::default();
+        decoration_info.insert("highlight", byte_ranges);
+
+        sources::print_src(
+            w,
+            contents_subset,
+            call_data.edition,
+            file_span,
+            cx,
+            &root_path,
+            Some(line_offset),
+            Some(highlight::DecorationInfo(decoration_info)),
+        );
+        write!(w, "</div></div>");
+
+        true
+    };
+
+    // The call locations are output in sequence, so that sequence needs to be determined.
+    // Ideally the most "relevant" examples would be shown first, but there's no general algorithm
+    // for determining relevance. Instead, we prefer the smallest examples being likely the easiest to
+    // understand at a glance.
+    let ordered_locations = {
+        let sort_criterion = |(_, call_data): &(_, &CallData)| {
+            // Use the first location because that's what the user will see initially
+            let (lo, hi) = call_data.locations[0].enclosing_item.byte_span;
+            hi - lo
+        };
+
+        let mut locs = call_locations.into_iter().collect::<Vec<_>>();
+        locs.sort_by_key(sort_criterion);
+        locs
+    };
+
+    let mut it = ordered_locations.into_iter().peekable();
+
+    // An example may fail to write if its source can't be read for some reason, so this method
+    // continues iterating until a write suceeds
+    let write_and_skip_failure = |w: &mut Buffer, it: &mut Peekable<_>| {
+        while let Some(example) = it.next() {
+            if write_example(&mut *w, example) {
+                break;
+            }
+        }
+    };
+
+    // Write just one example that's visible by default in the method's description.
+    write_and_skip_failure(w, &mut it);
+
+    // Then add the remaining examples in a hidden section.
+    if it.peek().is_some() {
+        write!(
+            w,
+            r#"<details class="rustdoc-toggle more-examples-toggle">
```

Comment:
Please add backslash at the end of each line (and not use `r#` I guess).

Reply:
Why?

Reply:
Is there a reason to not use `r#`? IMO the code is clearer this way. That way I don't need backslashes at end-of-line, and I don't need to escape strings. (I also saw `r#` used in several other places in the codebase.)

Reply:
> ~Why?~ Oh, to avoid extra whitespace in the generated HTML. Aren't we minifiying the HTML already? This seems silly to worry about, I think we should minify if we aren't already.

We don't and minifying HTML is a nightmare on so many levels.



> Is there a reason to not use `r#`? IMO the code is clearer this way. That way I don't need backslashes at end-of-line, and I don't need to escape strings. (I also saw `r#` used in several other places in the codebase.)

But that uses a lot more of character. We plan to use templates later but until then, please write code without `r#` if there are multiple lines (and end lines with backslash).